### ISSUE TITLE
[bug-arm32] The variable of type int returns an incorrect value

### ISSUE THREAD

**Issue by CoderBuck** [dart-lang/sdk#55080]
During runtime of `android-arm32-release.apk` (abiFilters "armeabi-v7a"), some `int` variables mysteriously return incorrect values. For example, assigning `1709049600000` to `final int time`, when displaying `time`, it might be `1708239506177`.

**Only appears in arm32 release mode.**

Here is a minimal reproducible demo, includes a `run.sh` file.
https://github.com/CoderBuck/bug_const

The expected value for `widget.time` is `1709049600000`, but some is wrong
<img width="350" alt="image" src="https://github.com/dart-lang/sdk/assets/17465838/7db7000d-510a-4334-982f-33928de1f7e4">

the `HomePage` code:
```dart
import 'package:flutter/material.dart';

class HomePage extends StatefulWidget {
  HomePage({super.key, required this.time});

  final int time;

  @override
  State<HomePage> createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Home')),
      body: Center(
        child: Column(
          children: [
            Text('widget.time: ${widget.time}'),
            Text('widget.time: ${widget.time}'),
            Text('widget.time: ${widget.time}'),
            Text('widget.time: ${widget.time}'),
          ],
        ),
      ), 
    );
  }
}

```

This bug is related to `const`. After removing the `const` keyword below, the bug disappears.
<img width="582" alt="image" src="https://github.com/dart-lang/sdk/assets/17465838/8ad12730-e11e-4b9c-a217-491f2fe17122">

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.19.1, on macOS 12.7.1 21G920 darwin-x64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] IntelliJ IDEA Community Edition (version 2023.2.5)
[✓] VS Code (version 1.78.2)
[✓] Connected device (3 available)
[✓] Network resources
```

**Comment by mraleph** [comment]
@CoderBuck please put some open source LICENSE of your choice (e.g. MIT, Apache, GPL, LGPL) in your repository otherwise we can't look at your code.

**Comment by CoderBuck** [comment]
@mraleph done

**Comment by mraleph** [comment]
I have tried reproducing it (using Flutter 3.19.1) - but unfortunately it does not reproduce using your repository. 

Can you upload the APK you get from the build somewhere? I can try doing a binary diff to see if I can spot the problem.

**Comment by CoderBuck** [comment]
Oh, it was my oversight.
I was able to reproduce the issue in flutter 3.13.6（Dart 3.1.3）, but not in 3.19.1.

I initially encountered this problem in 3.13.6, then switched to 3.19.1. However, I forgot to run flutter clean, which led me to believe that 3.19.1 also had the problem. In reality, 3.19.1 is fine.

**Comment by mraleph** [comment]
@CoderBuck I am going to close this then. For posterity 3.13.6 had a bug which caused incorrect code to be generated for `BoxInt64` instruction which was since fixed. The last fix was dcac60b67277ee20f9a12d43b3ca562a6b002957 which should be part of 3.19.x
